lib/linux_admin/network_interface/network_interface_rh.rb

```diff
@@ -0,0 +1,106 @@
+require 'ipaddr'
+require 'pathname'
+
+module LinuxAdmin
+  class NetworkInterfaceRH < NetworkInterface
+    IFACE_DIR = "/etc/sysconfig/network-scripts"
+
+    # @return [Hash<String, String>] Key value mappings in the interface file
+    attr_reader :interface_conf
+
+    # @param [String] Name of the network interface to manage
+    def initialize(interface)
+      super
+      @iface_file = Pathname.new(IFACE_DIR).join("ifcfg-#{@interface}")
+      reload
+    end
+
+    # Parses the interface configuration file into the @interface_conf hash
+    def reload
+      @interface_conf = {}
+      contents = File.read(@iface_file)
+
+      contents.split("\n").each do |line|
+        next if line =~ /^\s*#/
+
+        pair = line.split('=')
+        @interface_conf[pair[0]] = pair[1]
+      end
+    end
+
+    # Sets a static address for this network interface
+    # Accepts either IPv4 or IPv6 format addresses
+    #
+    # @param address [String] New static IP
+    # @param netmask [String] New sub-net mask
+    # @param gateway [String] New gateway address
+    # @raise [ArgumentError] if the parameter IP versions are not the same or an address
+    #   is malformed
+    def set_static_address(address, netmask, gateway)
+      begin
+        address_ip = IPAddr.new(address)
+        netmask_ip = IPAddr.new(netmask)
+        gateway_ip = IPAddr.new(gateway)
+      rescue IPAddr::InvalidAddressError => e
+        raise ArgumentError, "Bad address format: #{e.message}"
+      end
+
+      if address_ip.ipv4? && netmask_ip.ipv4? && gateway_ip.ipv4?
+        set_v4_address(address_ip, netmask_ip, gateway_ip)
+      else
+        raise ArgumentError, "Incompatible address formats provided for #{__method__}"
+      end
+      save_file
+    end
+
+    # Set up the interface to use DHCP
+    # Removes any previously set static networking information
+    def set_dhcp
+      @interface_conf["BOOTPROTO"] = "dhcp"
+      @interface_conf.delete("IPADDR")
+      @interface_conf.delete("NETMASK")
```

Comment:
An alternative to "NETMASK" is "PREFIX" which contains the CIDR version of the subnet mask.  You'll want to ensure that is also gone.